src/libstd/sys/unix/thread.rs

```diff
@@ -206,6 +207,29 @@ pub unsafe fn create(stack: uint, p: Thunk) -> rust_thread {
     native
 }
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+pub unsafe fn set_name(name: &str) {
+    // Using prctl() rather than pthread_setname_np(),
+    // because pthread_setname_np() wasn't added until glibc 2.12
+    // PR_SET_NAME since Linux 2.6.9
+    let cname = CString::from_slice(name.as_bytes());
+    prctl(15i32 /* = PR_SET_NAME */, cname.as_ptr() as u64, 0u64, 0u64, 0u64);
+}
+
+#[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]
+pub unsafe fn set_name(name: &str) {
+    // pthread_set_name_np() since almost forever on all BSDs
+    let cname = CString::from_slice(name.as_bytes());
+    pthread_set_name_np(pthread_self(), cname.as_ptr());
+}
+
+#[cfg(target_os = "macos")]
```

Comment:
Found it: [over here](https://www.opensource.apple.com/source/libpthread/libpthread-105.1.4/pthread/pthread.h) it says `pthread_setname_np()` is available since OS X 10.6 and iOS 3.2.